javascript/javascript3/week1/Giphy api/Giphyapi.js

```diff
@@ -0,0 +1,34 @@
+let inputValueapi = document.querySelector(".inputValueapi");
+let inputNumber = document.querySelector(".inputNumber");
+let butGiphyapli = document.querySelector(".butGiphyapli");
+let divImg = document.querySelector(".divImg");
+
+butGiphyapli.addEventListener("click", function () {
+  const apiUri = `https://api.giphy.com/v1/gifs/search?api_key=6oHY2zCDI0oLrTEEfdZic1OzwV0ezEa2&q=${inputValueapi.value}`;
+
+  async function giphyapi() {
+    const response = await fetch(apiUri);
+    const date = await response.json();
+
+    
+
+    inputNumber.addEventListener("onchang", giphyapi(fungifs));
```

Comment:
There are multiple issues with this event listener:
1. You probably meant to use the event [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) instead of`onchang` 🙂 
2. By adding the `(fungifs)` parameter, you're calling the `giphyapi` method immediately, passing the result of the method to the event listener. That means, every time your event is fired, the result of your method is being called (which fails, because the result is `undefined`). Instead of passing the result to the event listener, you'd want to pass a function. If you want to keep the parameter, it could be something like `("change", function() { giphyapi(fungifs) })`. Or you can make the `fungifs` variable available to the method in another way and pass the function name without parameters like `("change", giphyapi)`. It looks like `fungifs` is a function itself, why would you need to pass it in? 🤓 
3. You created an infinite loop by adding the event listener every time the `giphyapi` method is being called and calling the `giphyapi` over and over again whenever the event is fired 💣

Reply:
Here I wanted  when the user  changes the number of photos  he or she requested without pressing the button.